lib/network.go

```diff
@@ -5515,40 +5522,12 @@ func (txnData *DAOCoinLimitOrderMetadata) ToBytes(preSignature bool) ([]byte, er
 	data = append(data, EncodeUint256(txnData.ScaledExchangeRateCoinsToSellPerCoinToBuy)...)
 	data = append(data, EncodeUint256(txnData.QuantityToBuyInBaseUnits)...)
 	data = append(data, BoolToByte(txnData.CancelExistingOrder))
-	data = append(data, UintToBuf(uint64(len(txnData.MatchingBidsInputsMap)))...)
+	data = append(data, UintToBuf(uint64(len(txnData.MatchedBidsTransactors)))...)
 
-	// Maps MUST be iterated in sorted order or else the serialization of the txn
-	// will be non-deterministic. This is very bad because it results in the txn
-	// hash being different each time.
-	//
-	// We sort the keys based on the mainnet string encoding of the public key.
-	sortedBidderPkStrings := []string{}
-	for bidderPkIter, _ := range txnData.MatchingBidsInputsMap {
-		// You must make a copy of the iterator before taking a pointer of it.
-		// If you don't do this, you'll have one of the worst bugs you've ever
-		// had to deal with, where the underlying data of all your pointers is
-		// shifted under your feet. It's not needed here because we're not saving
-		// the pointer, but do it just to be safe.
-		bidderPk := bidderPkIter
-		sortedBidderPkStrings = append(sortedBidderPkStrings, PkToStringMainnet(bidderPk[:]))
-	}
-	for ii, pkStr := range sortedBidderPkStrings {
-		bidderPkBytes, _, err := Base58CheckDecode(pkStr)
-		if err != nil {
-			return nil, errors.Wrapf(err, "DAOCoinLimitOrderMetadata: "+
-				"Error serializing pubkey %v at index %v in MatchingBidsInputsMap",
-				pkStr, ii)
-		}
-		bidderPk := NewPublicKey(bidderPkBytes)
-		inputs, exists := txnData.MatchingBidsInputsMap[*bidderPk]
-		if !exists {
-			return nil, fmt.Errorf("DAOCoinLimitOrderMetadata: Error "+
-				"serializing inputs for pubkey %v at index %v. Map value "+
-				"did not exist. This should never happen", pkStr, ii)
-		}
-		data = append(data, bidderPk[:]...)
-		data = append(data, UintToBuf(uint64(len(inputs)))...)
-		for _, input := range inputs {
+	for _, transactor := range txnData.MatchedBidsTransactors {
```

Comment:
No need to worry about sorting by public key since since we're dealing with a slice instead of a map

Reply:
Couldn't the ordering of the slice be different resulting in a different txn hash?

Reply:
Hmm, given an existing slice, serializing and deserializing it to/from bytes won't change the order of elements. Same thing with serializing/deserializing to/from json. So we're safe in this scenario.

But if we ever need to reconstruct the same transaction from scratch (ex: based on input params from the API), then we could end up with a different ordering of elements. The place to address that is in the `CreateDAOCoinLimitOrderTxn` function which is where we construct this metadata type. I can make the change there, but will we ever hit this scenario?

Reply:
I'm not sure, but this ominous comment makes me think we should just double-check some consistent sorted ordering in the slice.
```
	// Maps MUST be iterated in sorted order or else the serialization of the txn
	// will be non-deterministic. This is very bad because it results in the txn
	// hash being different each time.
```